Painless Context Doc: Add field context example

mayya-sharipova opened the pull request:
relates to #34829

elasticmachine commented:
Pinging @elastic/es-core-infra

debadair commented on docs/painless/painless-contexts/painless-field-context.asciidoc:
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
+
+We will use two scripts to create two additional fields for each search hit.
+These script fields will output computed custom information.
+
```
In general, I recommend sticking with the present tense and avoiding "we" except for cases where "we" means "Elastic" (particularly in cases where you're making a recommendation). In this case, I'd probably edit this to read:

You can then use these example scripts to compute custom information and output it to two new fields.

debadair commented on docs/painless/painless-contexts/painless-field-context.asciidoc:
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
+These script fields will output computed custom information.
+
+The first script calculates the day of the week for the field `datetime`:
+
```
The callout below is fairly redundant. I'd remove the callout and edit the intro to read:

The first script gets the doc value for the `datetime` field and calls the `getDayOfWeek` function to determine the corresponding day of the week.

debadair commented on docs/painless/painless-contexts/painless-field-context.asciidoc:
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
+
+The second script calculates the number of actors. Actors' names are stored
+as an array in the `actors` field:
+
```
I'd say "as a text array" -- then you don't have to reitrate that `actors` is a text field in the callout. 

I'd  also change the :  to a . to be consistent with the punctuation above. (And the second sentence adds info, rather than really leading into the example.)

debadair commented on docs/painless/painless-contexts/painless-field-context.asciidoc:
```diff
@@ -26,3 +26,64 @@ a customized value for each document in the results of a query.
+    not available for this field. We parse `_source` to extract
+    `actors` and calculate their number. Note how `params['_source']['actors']`
+    is a list.
+
```
I'd rephrase this slightly:

By default, doc values are not available for text fields. However, you can still calculate the number of actors by extracting `actors` from  `_source`. Note that `params['_source']['actors']` is a list.

mayya-sharipova commented:
@debadair  Thank you for your review. I have addressed your comments the way you suggested. Can you please let me know if there is anything else that needs to be done.
